lms/static/scripts/frontend_apps/components/GradingCommentButton.tsx

```diff
@@ -73,9 +73,10 @@ function CommentPopover({
       <div className="flex flex-row-reverse space-x-2 space-x-reverse mt-3">
         <Button
           variant="primary"
-          onClick={e => {
-            onSubmit(e);
-            closePopover();
+          onClick={() => {
+            if (onSubmit()) {
+              closePopover();
```

Comment:
The popover is closed as soon as we know the form is valid, not after submitting succeeded, so this only prevents closing the popover of the form is not valid.